goth/runner/log_monitor.py

```diff
@@ -115,30 +114,53 @@ def _create_file_logger(config: LogConfig) -> logging.Logger:
     return logger_
 
 
-class LogEventMonitor(EventMonitor[LogEvent]):
+class PatternMatchingWaitableMonitor(WaitableEventMonitor[E]):
+    """A `WaitableEventMonitor` that can wait for events that match regex patterns."""
+
+    def event_str(self, event: E) -> str:
+        """Return the string associated with `event` on which to perform matching."""
+        return str(event)
+
+    async def wait_for_pattern(self, pattern: str, timeout: float = 1000) -> E:
```

Comment:
The timeout unit here seems inconsistent with `wait_for_event` which uses seconds (internally it's calling `Assertion#wait_for_result` which, in turn, calls `asyncio.wait_for`):
https://github.com/golemfactory/goth/blob/d130a47793b4303c3fae2dda345e85be59275fdf/goth/assertions/monitor.py#L312-L325

Reply:
Here `1000` means a thousand seconds, which serves as an approximation of infinity. But I guess I can just replace it with `None`.